File: server/src/protocol/StartStreamCommand.cpp
#include "StartStreamCommand.h"
#include "CommandVisitor.h"

using namespace std;
using namespace communication;

StartStreamCommand::StartStreamCommand()
        : Command(10, CommandType::START_STREAM)
{}

StartStreamCommand::~StartStreamCommand()
{}

vector<uint8_t> StartStreamCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    return frame;
}

std::string StartStreamCommand::getName()
{
    return string("StartStreamCommand");
}

void StartStreamCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void StartStreamCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);

    setDataSize(dataSize);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/StartStreamCommand.java
package wpam.mobile_client.protocol;
import java.util.ArrayList;

public final class StartStreamCommand extends Command
	{
		public StartStreamCommand()
		{
			super(10, CommandType.START_STREAM);
		}
		public final void close()
		{
			super.close();
		}

		@Override
		public  ArrayList<Integer> getFrameBytes()
		{
			initializeDataSize();

			ArrayList<Integer> frame = getHeader();
			frame.add(commandType_.getValue());

			return frame;
		}
		@Override
		public String getName()
		{
			return new String("StartStreamCommand");
		}
		@Override
		public void accept(CommandVisitor visitor)
		{
			visitor.visit(this);
		}

		@Override
		protected void initializeDataSize()
		{
			setDataSize(1);
		}
	}